ludwig/features/numerical_feature.py

```diff
@@ -132,6 +171,51 @@ def fit_transform_params(column: np.ndarray, backend: "Backend") -> dict:  # noq
 }
 
 
+def get_transformer(metadata, preprocessing_parameters):
+    return get_from_registry(
+        preprocessing_parameters.get("normalization", None),
+        numeric_transformation_registry,
+    )(**metadata)
+
+
+class _NumericalPreprocessing(torch.nn.Module):
+    def __init__(self, metadata: Dict[str, Any]):
+        super().__init__()
+        self.numeric_transformer = get_transformer(metadata, metadata["preprocessing"])
+
+    def forward(self, v: Union[List[str], torch.Tensor]):
+        if not isinstance(v, torch.Tensor):
+            raise ValueError(f"Unsupported input: {v}")
+        v = v.to(dtype=torch.float32)
+        return self.numeric_transformer.transform_inference(v)
+
+
+class _NumericalPostprocessing(torch.nn.Module):
+    def __init__(self, metadata: Dict[str, Any]):
+        super().__init__()
+        self.numeric_transformer = get_transformer(metadata, metadata["preprocessing"])
+        self.predictions_key = PREDICTIONS
+
+    def forward(self, preds: Dict[str, torch.Tensor]) -> Dict[str, Any]:
+        return {self.predictions_key: self.numeric_transformer.inverse_transform_inference(preds[self.predictions_key])}
```

Comment:
Discussed over Zoom. This is a good practice in general, but I don't feel comfortable having a single function that essentially does two completely different things based on a flag. If we were to go this route, I would rather break the transform and inverse transform into separate modules, but this may require some more design first.

Reply:
+1. Having separate functions seems totally sufficient and is quite clear.